src/timeout.rs

```diff
@@ -56,6 +56,18 @@ impl From<i32> for Timeout {
     }
 }
 
+impl From<Duration> for Timeout {
+    fn from(duration: Duration) -> Timeout {
+        if duration.is_zero() {
+            Timeout::Never
```

Comment:
I don't think it's a good idea to convert Duration `0` to `Never` just because the 0 value means never.
If you handover a parameter of type Duration then this behavior would be very unexpected.
What if we convert a timeout of Duration of `0ms` into a `None` timeout.

Reply:
But we have to return a ` Timeout` object, we can't return ` Option<Timeout>`. Then either we take the user at their word and return `Milliseconds(0)`  or we try to be smarter then them and return `Default`. What do you think ?

Reply:
oh yes, you're right

Reply:
I guess it's just weird API. then let's leave it like this. Thanks for the change, I'll just add a bit more documentation to make sure this is explained to the user.